device_amd64.go

```diff
@@ -0,0 +1,18 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+const (
+	rootBusPath = "/devices/pci0000:00"
+	pciBusMode  = 0220
+)
+
+var pciBusRescanFile = sysfsDir + "/bus/pci/rescan"
+
+func rescanPciBus() error {
```

Comment:
Since hot-add differs only on ARM, how about keeping the new files (`device_amd64.go`, `device_s390x.go`, *etc*), but having all non-ARM implementations call a generic handler like this:

```go
func rescanPciBus() error {
    return genericRescanPCIBus()
}
```

That would avoid the duplication.

Reply:
@jodh-intel thanks, if the implementation for rescanPciBus contains many lines of code, it is better to package them. but for only one line, keep it maybe better, as I fear of that more function call need more stack context switch which may be harmful to performance.